File: WebViewJavascriptInterafce/src/main/java/tech/thdev/webviewjavascriptinterface/view/main/presenter/MainContract.java
package tech.thdev.webviewjavascriptinterface.view.main.presenter;

import tech.thdev.kotlin_example_01.base.presenter.BasePresenter;
import tech.thdev.kotlin_example_01.base.presenter.BaseView;
import tech.thdev.webviewjavascriptinterface.webkit.CustomJavaScript;
import tech.thdev.webviewjavascriptinterface.webkit.listener.OnCustomJavaScriptListener;

/**
 * Created by Tae-hwan on 8/8/16.
 */

public interface MainContract {

    interface View extends BaseView<Presenter> {

        /**
         * Load url
         */
        void loadUrl(String url);

        /**
         * WebKeyword update.
         */
        void updateKeyword(String keyword);
    }

    interface Presenter extends BasePresenter<View>, OnCustomJavaScriptListener {

        CustomJavaScript getCustomJavaScript();
    }
}

File: app-kotlin/src/main/java/tech/thdev/app_kotlin/view/presenter/MainContract.kt
package tech.thdev.app_kotlin.view.presenter

import android.content.Context
import tech.thdev.base.presenter.BasePresenter
import tech.thdev.base.presenter.BaseView

/**
 * Created by tae-hwan on 18/12/2016.
 */

interface MainContract {

    interface View : BaseView {

        fun showStartOverlay()

        fun showObtainingPermissionOverlayWindow()
    }

    interface Presenter : BasePresenter<View> {

        fun startOverlayWindowService(context: Context)

        fun onOverlayResult(context: Context)
    }
}